src/librustdoc/html/templates/page.html

```diff
@@ -7,8 +7,7 @@
     <meta name="description" content="{{page.description}}"> {# #}
     <title>{{page.title}}</title> {# #}
     <script>if(window.location.protocol!=="file:") {# Hack to skip preloading fonts locally - see #98769 #}
-    for(f of "{{files.source_serif_4_regular}},{{files.fira_sans_regular}},{{files.fira_sans_medium}},{{files.source_code_pro_regular}},{{files.source_code_pro_semibold}}".split(",")) {# #}
-     document.write(`<link rel="preload" as="font" type="font/woff2" crossorigin href="{{static_root_path|safe}}${f}">`) {# #}
+    document.head.insertAdjacent("beforeend","{{files.source_serif_4_regular}},{{files.fira_sans_regular}},{{files.fira_sans_medium}},{{files.source_code_pro_regular}},{{files.source_code_pro_semibold}}".split(",").map(f=>`<link rel="preload" as="font" type="font/woff2" crossorigin href="{{static_root_path|safe}}${f}">`)) {# #}
```

Comment:
```suggestion
    document.head.insertAdjacent("beforeend","{{files.source_serif_4_regular}},{{files.fira_sans_regular}},{{files.fira_sans_medium}},{{files.source_code_pro_regular}},{{files.source_code_pro_semibold}}".split(",").map(f=>`<link rel="preload" as="font" type="font/woff2" crossorigin href="{{static_root_path|safe}}${f}">`)) {# #}
        document.head.insertAdjacentHTML("beforeend","{{files.source_serif_4_regular}},{{files.fira_sans_regular}},{{files.fira_sans_medium}},{{files.source_code_pro_regular}},{{files.source_code_pro_semibold}}".split(",").map(f=>`<link rel="preload" as="font" type="font/woff2" crossorigin href="{{static_root_path|safe}}${f}">`)) {# #}
```

Reply:
And I don't know how the template engine works, but

```
map(f=>`
```

looks a bit sus. Does it concat template strings properly?

Reply:
:+1:

Reply:
> And I don't know how the template engine works, but
> 
> ```
> map(f=>`
> ```
> 
> looks a bit sus. Does it concat template strings properly?

Checking.

Reply:
Seems so. I checked it as follows:

You need to have a local web server which you can use with:

```
python3 -m http.server
```

Then go to `http://localhost:8000/std/index.html` and check if the DOM contains the `preload` fonts. There were there so all good.

Reply:
Forgot to call `.join('')`...

Reply:
The \` is a javascript template string, so it should be around the whole parameter of all the `link` elements, not inside the map function.